src/steamship/agents/llms/openai.py

```diff
@@ -38,3 +38,52 @@ def complete(self, prompt: str, stop: Optional[str] = None) -> List[Block]:
         action_task = self.generator.generate(text=prompt, options=options)
         action_task.wait()
         return action_task.output.blocks
+
+
+# TODO(dougreid): should this be a an extension of a BaseOpenAI or similar? Or completely separate?
+class ChatOpenAI(ConversationalLLM):
+    """LLM that uses Steamship's OpenAI plugin to generate completions.
+
+    NOTE: By default, this LLM uses the `gpt-3.5-turbo` model. Valid model
+    choices are `gpt-3.5-turbo` and `gpt-4`.
+    """
+
+    generator: PluginInstance
+    client: Steamship
+
+    def __init__(
+        self, client, model_name: str = "gpt-4-0613", temperature: float = 0.4, *args, **kwargs
+    ):
+        """Create a new instance.
+
+        Valid model names are:
+         - gpt-4
+         - gpt-3.5-turbo
+        """
+        client = client
+        generator = client.use_plugin(
+            PLUGIN_HANDLE, config={"model": model_name, "temperature": temperature}
+        )
+        super().__init__(client=client, generator=generator, *args, **kwargs)
+
+    def complete(self, prompt: str, stop: Optional[str] = None) -> List[Block]:
+        options = {}
+        if stop:
+            options["stop"] = stop
+        action_task = self.generator.generate(text=prompt, options=options)
+        action_task.wait()
+        return action_task.output.blocks
+
+    def converse(self, messages: List[Block], tools: Optional[List[Tool]]) -> List[Block]:
+        # TODO(dougreid): this feels icky. find a better way?
```

Comment:
I get the ickyness here but fwiw can't think of anything better. To rubber duck it to GH in case it's useful:

1. The available tools is part of the data plane -- not prompt plane -- of this OAI model.
2. That means it needs to be passed in the `options` block to the `generate` method call.
3. To further the special-casey-ness, the thing passed to the options block also needs to be a special OAI-specific object.

The best suggestion I think I have would be to move `as_openai_function` into this class, since this class is related to OpenAI specifically, where as the tool base class isn't. But that doesn't seem to me something screaming to be done either.

Reply:
Interesting. The ickiness to me is the `File.create` bit (as we are taking blocks from another file and then creating a temp file with just those blocks here), solely for the purpose of calling `generate`. 

The tool conversion bit doesn't bother me as much.

Reply:
Ah, I missed that. Yeahhh. I'm OK with it -- we'll keep evolving this, and this works pretty great already having played with it a good bit